src/v2/Components/Modal/Modal.tsx

```diff
@@ -48,6 +49,9 @@ export class Modal extends React.Component<ModalProps> {
 
   close = () => {
     this.props.onClose()
+    if (this.props.hideCloseButton) {
+      window.history.back()
```

Comment:
This mostly works and would probably be fine! But in general we'll want to always use the router directly, versus indirectly.

You'll want to use our router hook and / or higher order component to do this. So down toward the bottom, something like:

```tsx
// instead of 
export default Modal

// do this:
const WrappedModal = props => {
  const router = useRouter() // from src/v2/Artsy/Router/useRouter.ts
  return <Modal {...props} router={useRouter}>
}

export default WrappedModal 
```
And then here, where you access the router through window, you'll want to use the router:
```tsx
this.props.router.back()
```

Reply:
I'd like to convert the `Modal` and `ModalContainer` classes to functional components and use a router to go back to the previous page. We may also want to navigate users back to the landing page for Viewing Rooms in case they access a viewing room from a URL instead of navigating there.

Since the (second) launch date for viewing rooms is tomorrow, I'd like to merge this, test it, and iterate.